addons/payment/models/payment_transaction.py

```diff
@@ -0,0 +1,943 @@
+            f"{self.acquirer_id.id}:\n{pprint.pformat(processing_values)}",
+        )
+
+        # Render the html form for the redirect flow if available
+        if self.acquirer_id.redirect_form_view_id:
+            rendering_values = self._get_specific_rendering_values(processing_values)
+            redirect_form_html = self.acquirer_id.redirect_form_view_id._render(
+                rendering_values, engine='ir.qweb'
+            )
+            processing_values.update(redirect_form_html=redirect_form_html)
+
+        return processing_values
+
+    def _get_specific_processing_values(self, _processing_values):
+        """ Return a dict of acquirer-specific values used to process the transaction.
+
+        For an acquirer to add its own processing values, it must overwrite this method and return a
+        dict of acquirer-specific values based on the generic values returned by this method.
+        Acquirer-specific values take precedence over those of the dict of generic processing
+        values.
+
+        :param dict _processing_values: The generic processing values of the transaction
+        :return: The dict of acquirer-specific processing values
+        :rtype: dict
+        """
+        return dict()
+
+    def _get_specific_rendering_values(self, _processing_values):
+        """ Return a dict of acquirer-specific values used to render the redirect form.
+
+        For an acquirer to add its own rendering values, it must overwrite this method and return a
+        dict of acquirer-specific values based on the processing values (acquirer-specific
+        processing values included).
+
+        :param dict _processing_values: The processing values of the transaction
+        :return: The dict of acquirer-specific rendering values
+        :rtype: dict
+        """
+        return dict()
+
+    def _send_payment_request(self):
+        """ Request the provider of the acquirer handling the transactions to execute the payment.
+
+        For an acquirer to support tokenization, it must override this method and call it to log the
+        'sent' message, then request a money transfer to its provider.
+
+        :return: None
+        """
+        self._log_sent_message()
+    
+    @api.model
+    def _handle_feedback_data(self, provider, data):
+        """ Match the transaction with the feedback data, update its state and return it.
+
+        :param str provider: The provider of the acquirer that handled the transaction
+        :param dict data: The feedback data sent by the provider
+        :return: The transaction if found
+        :rtype: recordset of `payment.transaction`
+        """
+        tx = self._get_tx_from_feedback_data(provider, data)
+        if tx:
+            tx._process_feedback_data(data)
+            tx._execute_callback()
+        else:
+            pass  # The transaction might not be recorded in Odoo in some acquirer-specific flows
+        return tx
```

Comment:
is the returned value necessary?  I don't see it used anywhere
the `pass` should probably be removed to at least return an empty recordset ?

Reply:
I think that I had a use for it once but not anymore.
I'll leave this comment open and see if any other acquirer needs it once they are all merged.